src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -35,36 +36,86 @@ const SearchAndFilter: React.FC<filterProps> = ({data, columns, filterTypes}) =>
         }
       })
     }
-    setIsLoading(false)
-  }, [data, filterTypes, isLoading])
-  
-  
-  const filterData = useCallback((matchingItems:object[]) => {
-    let result:object[] = matchingItems
-    if (filterTypes) {
+  }, [allData, filterTypes])
+
+  const createFilterOptions = (filterType: string) => {
+    const availableOptions = determineAvailableFilterValues(filterType)
+
+    const options = availableOptions.map((option, i) => {
+        if (availableOptions.indexOf(option) === i) {
+          return <option value={option.toString()}>{option}</option>
+        }
+      })
+
+    return (
+      <>
+      <label htmlFor={`${filterType}-selector`}>{filterType}</label>
+      <select 
+        id={`${filterType}-selector`} 
+        onChange={(event) => {
+          dispatch({type: filterType, value: event.target.value})
+          messageIfSearchFailed()
+        }}
+        value={selectedFilters[filterType]}
+      >
+        <option value="">All</option>
+        {options}
+      </select>
+      </>
+    )
+  }
+
+  const filterData = useCallback(() => {
+    let result:Array<{id:string}> = []
+    
+    const someFilterIsSelected = ():boolean => {
+      return Object.values(selectedFilters).some(filter => filter !== 'All')
+    }
+    
+    if (filterTypes && someFilterIsSelected()) {
       filterTypes.forEach(filter => {
-        if (selectedFilters[filter] !== 'All' && filter !== 'search') {
-          result = result.filter((item:stringKeyOptions) => item[filter].includes(selectedFilters[filter]))
+        if (selectedFilters[filter] !== 'All') {
+          const dataToSearch = result.length > 0 ? result : allData
+          result = dataToSearch.filter((item:stringKeyOptions) => {
+            return item[filter].includes(selectedFilters[filter])
+          })
         } 
       })
     }
-    return result 
-  }, [filterTypes, selectedFilters])
-  
-  const limitSelection = useCallback(() => {
-    let matchingItems:object[] = data
-    matchingItems = filterData(matchingItems)
-    setFilteredData(matchingItems)
-  }, [data, filterData])
 
-  useEffect(() => {
-    limitSelection()
-    console.log('run')
-  }, [limitSelection, selectedFilters])
+    const resultIds = result.map(item => item.id)
+    setMatchingFilterIds(resultIds)
+  }, [allData, filterTypes, selectedFilters])
 
+  const cleanDataForSearch = (item: object):any[] => {
```

Comment:
I've done my best to only use `any[]` when allowing for other datasets to be used by this component